internal/graph/graph.go

```diff
@@ -293,15 +377,87 @@ func (g *ConnectedObjectGraph) findIngressesWithTargetRewrite(
 		var res []*RelationshipIngress
 		for _, child := range t.Union.GetChild() {
 			// we recurse through each child rewrite
-			childResults, err := g.findIngressesWithTargetRewrite(target, source, child, visited)
+			childResults, err := g.findIngressesWithTargetRewrite(target, source, child, visited, findIngressOption)
 			if err != nil {
 				return nil, err
 			}
 			res = append(res, childResults...)
 		}
 		return res, nil
-	case *openfgapb.Userset_Intersection, *openfgapb.Userset_Difference:
-		return nil, ErrNotImplemented
+	case *openfgapb.Userset_Intersection:
+
+		if findIngressOption == resolveAnyIngress {
+			child := t.Intersection.GetChild()[0]
+
+			childresults, err := g.findIngressesWithTargetRewrite(target, source, child, visited, findIngressOption)
+			if err != nil {
+				return nil, err
+			}
+
+			for _, childresult := range childresults {
+				childresult.Condition = RequiresFurtherEvalCondition
+			}
+
+			return childresults, nil
+		}
+
+		var ingresses []*RelationshipIngress
+		for _, child := range t.Intersection.GetChild() {
+
+			res, err := g.findIngressesWithTargetRewrite(target, source, child, visited, findIngressOption)
+			if err != nil {
+				return nil, err
+			}
+
+			ingresses = append(ingresses, res...)
+		}
+
+		if len(ingresses) > 0 {
+			ingresses[0].Condition = RequiresFurtherEvalCondition
+		}
+
+		return ingresses, nil
+	case *openfgapb.Userset_Difference:
+
+		if findIngressOption == resolveAnyIngress {
+			child := t.Difference.GetBase()
+
+			childresults, err := g.findIngressesWithTargetRewrite(target, source, child, visited, findIngressOption)
+			if err != nil {
+				return nil, err
+			}
+
+			for _, childresult := range childresults {
+				childresult.Condition = RequiresFurtherEvalCondition
+			}
+
+			return childresults, nil
+		}
+
+		var ingresses []*RelationshipIngress
+
+		baseRewrite := t.Difference.GetBase()
+
+		baseIngresses, err := g.findIngressesWithTargetRewrite(target, source, baseRewrite, visited, findIngressOption)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(baseIngresses) > 0 {
+			baseIngresses[0].Condition = RequiresFurtherEvalCondition
+		}
+
+		ingresses = append(ingresses, baseIngresses...)
+
+		subtractRewrite := t.Difference.GetSubtract()
+
+		subIngresses, err := g.findIngressesWithTargetRewrite(target, source, subtractRewrite, visited, findIngressOption)
+		if err != nil {
```

Comment:
Why aren't we setting RequiresFurtherEvalCondition if len(subIngresses) > 0

Reply:
Same comment as in https://github.com/openfga/openfga/pull/797#discussion_r1245690439. This needs more special resolution behavior, and we aren't well equipped to do so at this time.